### PR TITLE
Adding ListQueues in History and Admin service

### PR DESCRIPTION
**Pull request by prathyushpv**
**What changed?**
Adding ListQueues API to history service and admin service.

**Why?**
Operators will be using these APIs to get the list of DLQs.
A command will be added to tdbg to call this API.

**How did you test it?**
Unit tests with 100% coverage.

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: proto/internal/temporal/server/api/historyservice/v1/request_response.proto [hunk 1]
```diff
@@ -776,6 +776,16 @@ message DeleteDLQTasksResponse {
...
 }
 
+message ListQueuesRequest {
+    int32 queue_type = 1;
+    int32 page_size = 2;
+    bytes next_page_token = 3;
+}
+
+message ListQueuesResponse {
+    repeated string queue_names = 1;
```
Comment (MichaelSnowden) [inline proto/internal/temporal/server/api/historyservice/v1/request_response.proto:786]: Won't clients need the next page token here?

File: service/history/api/listqueues/listqueuestest/apitest.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	"go.temporal.io/server/service/history/api/listqueues"
+	"go.temporal.io/server/service/history/consts"
+	"go.temporal.io/server/service/history/tasks"
+)
+
+// TestInvoke is a library test function intended to be invoked from a persistence test suite. It works by
+// enqueueing a task into the DLQ and then calling [getdlqtasks.Invoke] to verify that the right task is returned.
+func TestInvoke(t *testing.T, manager persistence.HistoryTaskQueueManager) {
+	ctx := context.Background()
+	t.Run("HappyPath", func(t *testing.T) {
```
Comment (MichaelSnowden) [inline service/history/api/listqueues/listqueuestest/apitest.go:44]: Could we do a paginated query on the happy path? For example, add 3 queues, query through with a page size of 2 and verify that we get all 3 once the response's next page token is empty?

**Review by MichaelSnowden — COMMENTED** [review]
This looks perfect except for the missing page token in the response. I think we need that for clients to paginate properly

File: service/history/api/listqueues/listqueuestest/apitest.go (hunk 2 above)
Comment (prathyushpv) [reply service/history/api/listqueues/listqueuestest/apitest.go:45]: Yeah added that to this test.

File: proto/internal/temporal/server/api/historyservice/v1/request_response.proto (hunk 1 above)
Comment (prathyushpv) [reply proto/internal/temporal/server/api/historyservice/v1/request_response.proto:786]: Yes. Added the token.

**Review by MichaelSnowden — APPROVED** [review]

**Pull request merged**